$OpenBSD: patch-exopen_c,v 1.1 2001/07/26 21:37:31 rohee Exp $

In case of maildir format delivery, actually deliver in the right directory.

--- src/exopen.c.orig	Thu Jul 26 23:28:10 2001
+++ src/exopen.c	Thu Jul 26 23:28:59 2001
@@ -47,7 +47,7 @@ in:	if(flags&doMAILDIR)
 	   ultstr(0,(long)thepid,dot+1);
 	   *(dot=strchr(p,'\0'))='_';
 	   *(++dot+1)='.';
-	   strlcat(dot+2,hostname(),end-dot);
+	   strlcpy(dot+2,hostname(),end-dot-2);
 	 }
 	else
 	 { p=ultoan((long)t,dot+1);
